src/components/BooksForm.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react'; // eslint-disable-line no-unused-vars
+import { CREATE_BOOK } from '../actions'
+import { connect } from 'react-redux'
+
+const categories = ["Action", "Biography", "History", "Horror", "Kids", "Learning", "Sci-Fi"]
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      categorie: categories[0]
+    }
+  }
+  handleChange({ title, categorie }) {
+    if (title) {
+      this.setState({ title: title.value })
+    }
```

Comment:
Hello, this is fine, but let me suggest a simpler implementation and of course this is OPTIONAL.
```
handleChange = (e) => {
    this.setState({
      [e.target.name] : e.target.value
    })
  }
```
and the input can be:

```
<input
          type="text"
          //onChange={(e) => this.handleChange({ title: e.target })}
          onChange={this.handleChange}
          name="title"
          value={this.state.title}
        />
```
this implementation is using the name to track the event and update the state.
- this way no matter the amount of fields you have, the handleChange remains the same, you just have to give the input a name that matches what you have in store for it.

Reply:
and if you don't like that approach, there is no need to have multiple `setState`, it should look something like this:
```
handleChange(event) {
  const name = event.target.name;

  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}
```